test/collector.test.ts

```diff
@@ -0,0 +1,351 @@
+            expect(await testToken.balanceOf(remainder.address)).to.equal(3);
+        });
+
+        it('Should fail when token balance > = remainder', async () => {
+            const { collector, testToken, utilWallet } = await loadFixture(
+                prepareAllFixture
+            );
+            await testToken.mint(4, collector.address);
+
+            await expect(
+                collector.updateRemainderAddress(utilWallet.address)
+            ).to.be.revertedWith('there is balance to share');
+        });
+
+        it('Should fail when is called by an address that is not the owner', async () => {
+            const { collector, utilWallet } = await loadFixture(
+                prepareAllFixture
+            );
+            const externallyLinkedCollector = collector.connect(
+                utilWallet.address
+            );
+            await expect(
+                externallyLinkedCollector.updateRemainderAddress(
+                    utilWallet.address
+                )
+            ).to.be.revertedWith('only owner can call this');
+        });
+    });
+
+    describe('getBalance', () => {
+        it('Should return 0 if the contract has been just deployed', async () => {
+            const { collector } = await loadFixture(prepareAllFixture);
+            await expect(await collector.getBalance()).to.equal(0);
+        });
+
+        it('Should return 100 after that value has been minted', async () => {
+            const { collector, testToken } = await loadFixture(
+                prepareAllFixture
+            );
+            await testToken.mint(100, collector.address);
+            await expect(await collector.getBalance()).to.equal(100);
+        });
+    });
+
+    describe('withdraw', () => {
+        it('Should withdraw', async () => {
+            const { collector, testToken, partners } = await loadFixture(
+                prepareAllFixture
+            );
+
+            await testToken.mint(100, collector.address);
+            await collector.withdraw();
+
+            expect(await testToken.balanceOf(partners[0].beneficiary)).to.equal(
+                FIRST_PARTNER_SHARE
+            );
+            expect(await testToken.balanceOf(partners[1].beneficiary)).to.equal(
+                SECOND_PARTNER_SHARE
+            );
+            expect(await testToken.balanceOf(partners[2].beneficiary)).to.equal(
+                THIRD_PARTNER_SHARE
+            );
+            expect(await testToken.balanceOf(partners[3].beneficiary)).to.equal(
+                FOURTH_PARTNER_SHARE
+            );
+            expect(await testToken.balanceOf(collector.address)).to.equal(0);
+        });
+
+        it('Should fail when no revenue to share', async () => {
+            const { collector, testToken } = await loadFixture(
+                prepareAllFixture
+            );
+            await testToken.mint(1, collector.address);
+            await expect(collector.withdraw()).to.be.revertedWith(
```

Comment:
We shell add this as a formatting rule, but until then, always add a space before await and return statements for clarity

Reply:
Done.